### PR TITLE
Revert wraparound semantics for reading from buffers

### PR DESCRIPTION
**Pull request by blishko**
## Description

Recently, we have changed the semantics of reading beyond index `2^256` from a buffer. Previously, any read beyond buffer length would return all zeroes. However, to match the semantics of SMT, we changed that so that if reading from multiple consecutive indices would overflow the index (at `2^256`), the indices would wrap around and start from `0` again. Unfortunately, it turns out that the interpreters (such as evmtool) do not behave this way, they follow the original "just return all zeroes" strategy.

In order to match the behaviour of real interpreters, we revert the recent changes and return to the original behaviour.

This also introduces the problem of incompatibility with our SMT encoding, which should be mitigated by modifying the fuzzer to not generate these corner cases.

### CODE REVIEW

File: test/test.hs [hunk 1]
```diff
@@ -736,13 +741,8 @@ tests = testGroup "hevm"
           s = Expr.simplify e
         equal <- checkEquiv e s
         assertEqualM "Must be equal" True equal
-    , test "copy-slice-overflowing-into-too-short-buffer" $ do
```
Comment (blishko) [inline test/test.hs:739]: This test was basically a duplicate of copy-slice-overflow-beyond-source-into-nonempty
